scylla/src/client/caching_session.rs

```diff
@@ -247,9 +247,129 @@ where
     }
 }
 
+const DEFAULT_MAX_CAPACITY: usize = 128;
+
+/// [CachingSessionBuilder] is used to create new [CachingSession] instances.
+///
+/// # Example
+///
+/// ```
+/// # use scylla::client::session::Session;
+/// # use scylla::client::session_builder::SessionBuilder;
+/// # use scylla::client::caching_session::{CachingSession, CachingSessionBuilder};
+/// # async fn example() -> Result<(), Box<dyn std::error::Error>> {
+/// let session: Session = SessionBuilder::new()
+///     .known_node("127.0.0.1:9042")
+///     .build()
+///     .await?;
+/// let caching_session: CachingSession = CachingSessionBuilder::new(session)
+///     .max_capacity(2137)
+///     .build();
+/// # Ok(())
+/// # }
+/// ```
+pub struct CachingSessionBuilder<S = RandomState>
+where
+    S: Clone + BuildHasher,
+{
+    session: Session,
+    max_capacity: usize,
+    hasher: S,
+}
+
+impl CachingSessionBuilder<RandomState> {
+    /// Wraps a [Session] and creates a new [CachingSessionBuilder] instance,
+    /// which can be used to create a new [CachingSession].
+    ///
+    pub fn new(session: Session) -> Self {
+        Self {
+            session,
+            max_capacity: DEFAULT_MAX_CAPACITY,
+            hasher: RandomState::default(),
+        }
+    }
+}
```

Comment:
So the flow to create session with custom hasher using this is to first create builder with `RandomState` and then change it to a builder with different hasher using`with_hasher`?

Reply:
Yes, with small difference: the method is called `hasher()`.